src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php

```diff
@@ -513,17 +513,15 @@ public function parse($value, $type = self::TYPE_DOUBLE, &$position = 0)
             return false;
         }
 
-        $groupSep = $this->getAttribute(self::GROUPING_USED) ? ',' : '';
-
-        // Any string before the numeric value causes error in the parsing
-        if (preg_match("/^-?(?:\.\d++|([\d{$groupSep}]++)(?:\.\d*+)?)/", $value, $matches)) {
+        $groupingMatch = $this->getAttribute(self::GROUPING_USED) ? '|(?P<grouping>\d++(,{1}\d+)++(\.\d*+)?)' : '';
+        if (preg_match("/^-?(?:\.\d++{$groupingMatch}|\d++(\.\d*+)?)/", $value, $matches)) {
```

Comment:
I have added quite a few test cases to make sure no behaviour was broken by changes introduced in this code block: [AbstractNumberFormatterTest::testParse()]( https://github.com/symfony/symfony/pull/28977/files#diff-542534c8af70bcf20dd77bededfdb7c3R669)
Main change I noticed is that `-123,,456` didn't use to return a result, while now it parses as `-123`.